examples/nft-nd-ns/ns.go

```diff
@@ -0,0 +1,655 @@
+func getRecord(ctx storage.Context, tokenId []byte, name string, typ RecordType) string {
+	recordKey := getRecordKey(tokenId, name, typ)
+	record := storage.Get(ctx, recordKey)
+	if record == nil {
+		return ""
+	}
+	return record.(string)
+}
+
+// putRecord stores domain record.
+func putRecord(ctx storage.Context, tokenId []byte, name string, typ RecordType, record string) {
+	recordKey := getRecordKey(tokenId, name, typ)
+	storage.Put(ctx, recordKey, record)
+}
+
+// getRecordKey returns key used to store domain records.
+func getRecordKey(tokenId []byte, name string, typ RecordType) []byte {
+	recordKey := append([]byte{prefixRecord}, getTokenKey(tokenId)...)
+	recordKey = append(recordKey, getTokenKey([]byte(name))...)
+	return append(recordKey, []byte{byte(typ)}...)
+}
+
+// isValid returns true if the provided address is a valid Uint160.
+func isValid(address interop.Hash160) bool {
+	return address != nil && len(address) == 20
+}
+
+// checkCommittee panics if the script container is not signed by the committee.
+func checkCommittee() {
+	committee := neo.GetCommittee()
+	if committee == nil {
+		panic("failed to get committee")
+	}
+	l := len(committee)
+	committeeMultisig := contract.CreateMultisigAccount(l-(l-1)/2, committee)
+	if committeeMultisig == nil || !runtime.CheckWitness(committeeMultisig) {
+		panic("not witnessed by committee")
+	}
+}
+
+// checkFragment validates root or a part of domain name.
+func checkFragment(v string, isRoot bool) bool {
+	maxLength := 32
+	if isRoot {
+		maxLength = maxRootLength
+	}
+	if len(v) == 0 || len(v) > maxLength {
+		return false
+	}
+	c := v[0]
+	if isRoot {
+		if !(c >= 'a' && c <= 'z') {
+			return false
+		}
+	} else {
+		if !isLiteral(c) {
+			return false
+		}
+	}
+	for i := 1; i < len(v); i++ {
+		if !isLiteral(v[i]) {
+			return false
+		}
+	}
+	return true
+}
+
+// isLiteral checks whether provided char is a lowercase letter or a number.
+func isLiteral(c uint8) bool {
+	return c >= 'a' && c <= 'z' || c >= '0' && c <= '9'
+}
+
+// splitAndCheck splits domain name into parts and validates it.
+func splitAndCheck(name string, allowMultipleFragments bool) []string {
+	l := len(name)
+	if l < minDomainNameLength || maxDomainNameLength < l {
+		return nil
+	}
+	fragments := std.StringSplit(name, ".")
+	l = len(fragments)
+	if l < 2 {
+		return nil
+	}
+	if l > 2 && !allowMultipleFragments {
+		return nil
+	}
+	for i := 0; i < l; i++ {
+		if !checkFragment(fragments[i], i == l-1) {
+			return nil
+		}
+	}
+	return fragments
+}
+
+// checkIPv4 checks record on IPv4 compliance.
+func checkIPv4(data string) bool {
+	l := len(data)
+	if l < 7 || 15 < l {
+		return false
+	}
+	fragments := std.StringSplit(data, ".")
+	if len(fragments) != 4 {
+		return false
+	}
+	numbers := make([]int, 4)
+	for i, f := range fragments {
+		if len(f) == 0 {
+			return false
+		}
+		number := std.Atoi10(f) // TODO: add bounds check?  (C# code: byte number = byte.Parse(fragment);)
```

Comment:
We sure need something for `number > 255`.

Reply:
We should panic here, fixed.